### PR TITLE
Added custom render texture template in core

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Add a shader template to create custom render texture shaders:

image

By default the template code generate this kind of texture:
image

Here's an example of the generated code:

```CG
Shader "CustomRenderTexture/Test2"
{
    Properties
    {
        _Color ("Color", Color) = (1,1,1,1)
        _MainTex("InputTex", 2D) = "white" {}
     }

     SubShader
     {
        Blend One Zero

        Pass
        {
            Name "Test2"

            CGPROGRAM
            #include "UnityCustomRenderTexture.cginc"
            #pragma vertex CustomRenderTextureVertexShader
            #pragma fragment frag
            #pragma target 3.0

            float4      _Color;
            sampler2D   _MainTex;

            float4 frag(v2f_customrendertexture IN) : COLOR
            {
                float2 uv = IN.localTexcoord.xy;
                float4 color = tex2D(_MainTex, uv) * _Color;

                // TODO: Replace this by actual code!
                uint2 p = uv.xy * 256;
                return countbits(~(p.x & p.y) + 1) % 2 * float4(uv, 1, 1) * color;
            }
            ENDCG
        }
    }
}
```

---
### Testing status
I simply created a new CRT shader, assigned it to material, and assigned this material in a CRT.

### CODE REVIEW

**Review by JulienIgnace-Unity — APPROVED** [review]

**Review by sebastienlagarde — APPROVED** [review]
Please add this addition in the what's new of hdrp (I know it is in core but nobody will read it :) )

**Comment by alelievr** [comment]
Added @Vic-Cooper for the what's new :)

File: com.unity.render-pipelines.high-definition/Documentation~/whats-new-12.md [hunk 1]
```diff
@@ -131,6 +131,10 @@ HDRP 12.0 includes the following physical camera improvements:
 Improved the quality of the physically-based Depth Of Field.
 ![](Images/HDRPFeatures-BetterDoF.png)
 
+### Custom Render Texture shader
+
+A new menu is available to create a shader specifically formatted for [Custom Render Textures](https://docs.unity3d.com/Manual/class-CustomRenderTexture.html) at **Assets > Create > Shader > Custom Render Texture**. Note that you still need to create a material from this shader and assign it to the custom render texture field **Material**.
```
Comment (Vic-Cooper) [inline com.unity.render-pipelines.high-definition/Documentation~/whats-new-12.md:136]: This HDRP version includes a new shader specifically formatted for [Custom Render Textures](https://docs.unity3d.com/Manual/class-CustomRenderTexture.html) in **Assets > Create > Shader > Custom Render Texture**. To use this shader, create a new Material and assign it to the Custom Render Texture's **Material** field.

**Review by Vic-Cooper — COMMENTED** [review]
Tweaked the what's new page.

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**